src/tiramisu_core.cpp

```diff
@@ -7810,4 +7811,98 @@ void tiramisu::buffer::tag_gpu_register() {
     set_auto_allocate(false);
 }
 
+std::string get_rank_string_type(tiramisu::rank_t rank_type)
+{
+    if (rank_type == rank_t::r_sender)
+        return "r_sender";
+    else
+        return "r_receiver";
+}
+
+std::vector<std::string> computation::get_static_and_dynamic_dimensions_names()
```

Comment:
Can you use the function

std::vector<std::string> computation::get_iteration_domain_dimension_names()

instead of creating this one?

The current function fires an error if a name does not exist but I'm ok with you changing it to behave like your new function (i.e., it generates a new name if a name does not exist).

Reply:
I created this one because the existing function returns the names of dimensions of the iteration_domain, not trimmed_time_processor_domain.

Reply:
Then, can you call something like

computation::get_trimmed_time_space_domain_dimension_names()

This name is close to computation::get_iteration_domain_dimension_names() which we already have. Maybe in the documentation you can talk about the difference between the two functions.